src/core/algorithms/od/fastod/fastod.cpp

```diff
@@ -0,0 +1,304 @@
+#include "fastod.h"
+
+#include <algorithm>
+#include <iostream>
+#include <utility>
+
+#include <boost/unordered/unordered_map.hpp>
+#include <easylogging++.h>
+
+#include "config/equal_nulls/option.h"
+#include "config/names_and_descriptions.h"
+#include "config/option_using.h"
+#include "config/tabular_data/input_table/option.h"
+#include "config/time_limit/option.h"
+#include "single_attribute_predicate.h"
+#include "stripped_partition.h"
+
+namespace algos::fastod {
+
+Fastod::Fastod() : Fastod(0u) {}
+
+Fastod::Fastod(config::TimeLimitSecondsType time_limit_seconds)
+    : Algorithm({}), time_limit_seconds_(time_limit_seconds) {
+    PrepareOptions();
+}
```

Comment:
```suggestion
Fastod::Fastod() : Algorithm({}) {
    PrepareOptions();
}
```
We don't need to define second constructor with time limit param, because setting values for registered options is done by the algo factory (using corresponding option `config::TimeLimitSecondsOpt`). Note, that If the option was not explicitly passed to the map with options, then the default value of the option `TimeLimitSecondsOpt` will be used.

Reply:
Did it.